### PR TITLE
feat-depricate-usage-worker

### PR DESCRIPTION
**Pull request by lohanidamodar**
## What does this PR do?

Depricates usage worker and appwrite connects directly to statsd

## Test Plan

Existing tests

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

YES

### CODE REVIEW

File: app/workers/functions.php [hunk 1]
```diff
@@ -477,21 +475,26 @@ public function execute(string $trigger, string $projectId, string $executionId,
...
+
+            $tags = ",project={$projectId},version=".App::getEnv('_APP_VERSION', 'UNKNOWN');
+
+            // the global namespace is prepended to every key (optional)
+            $statsd->setNamespace('appwrite.usage');
+
+            $statsd->increment('executions.all'.$tags.',functionId='.$function->getId().',functionStatus='.$functionStatus);
+            $statsd->count('executions.time'.$tags.',functionId='.$function->getId(), $functionExecutionTime);
+
+            if($storage >= 1) {
```
Comment (TorstenDittmann) [inline app/workers/functions.php:494]: For what do we need this, if the condition will never trigger? 🙂

Comment (lohanidamodar) [reply app/workers/functions.php:494]: something i missed :D

File: src/Appwrite/Statsd/Statsd.php [hunk 2]
```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace Appwrite\Statsd;
+
+use Utopia\App;
+
+class Statsd
```
Comment (eldadfux) [inline src/Appwrite/Statsd/Statsd.php:7]: I think a better name would be `Stats` as this is an Appwrite specific wrapper. Would also allow to decouple from StatsD easily in the future.

File: src/Appwrite/Statsd/Statsd.php [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    {
+        $this->params[$key] = $value;
+
+        return $this;
+    }
+
+    /**
+     * Save to statsd.
+     */
+    public function save(): void
```
Comment (eldadfux) [inline src/Appwrite/Statsd/Statsd.php:45]: `save` is a bit misleading. The data is being submitted by UDP not really stored anywhere. `submit` would probably be a better name.

File: src/Appwrite/Statsd/Statsd.php [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+     */
+    public function setParam(string $key, $value): self
+    {
+        $this->params[$key] = $value;
+
+        return $this;
+    }
+
+    /**
+     * Save to statsd.
```
Comment (eldadfux) [inline src/Appwrite/Statsd/Statsd.php:43]:
```suggestion
     * Submit data to StatsD.
```

**Review by eldadfux — CHANGES_REQUESTED** [review]

**Pull request merged**